added a fancy class, hopefully it works better?

Sable-20 opened the pull request:
i did a lot of fancy things with textwrapping and making the custom command, if it doesnt work this will be hell since i basically didnt comment 99% ofit

Sable-20 committed:
added a fancy class, hopefully it works better?

i did a lot of fancy things with textwrapping and making the custom command, if it doesnt work this will be hell since i basically didnt comment 99% ofit

sepehrdaddev commented:
@Sable-20, Thank you so much for your time and effort that you put in to this PR.
After testing this PR, the changes that were applied completely broke the script.
apart from that it was a bit over engineered, that's why I will close this PR.

Sable-20 committed:
fixed up some tabs it should work now

Sable-20 committed:
fixed tabs again

pwnfoo commented on wordlistctl.py:
```diff
@@ -245,224 +242,235 @@ def lst_func(parser: argparse.ArgumentParser) -> None:
-		sys.stderr.write(("Error: %s" % message) + "\n")
-		sys.exit(2)
+    def __init__(self, *args, width=78,
+                 **kwargs):  # this is basically just to get the variables for the screen, a bit like writing characters to the screen when youre making a kernel, place a character and then increase the y coordinates by x pixles
```
I recommend using [docstrings](https://www.python.org/dev/peps/pep-0257/) for comments like this.

Sable-20 replied:
never

pwnfoo commented on wordlistctl.py:
```diff
@@ -245,224 +242,235 @@ def lst_func(parser: argparse.ArgumentParser) -> None:
+        super(ArgumentParser, self).add_argument(*args, **kwargs)
+        argument = {key: kwargs[key] for key in kwargs}
+
+        if (len(args) == 0 or (len(args) == 1 and isinstance(args[0], str) and not args[0].startswith("-"))):
```
Drop all brackets for if. 

![image](https://user-images.githubusercontent.com/9546091/110033820-e3137380-7d5f-11eb-9eed-9a3dbf338a04.png)

Sable-20 replied:
done

pwnfoo commented on wordlistctl.py:
```diff
@@ -245,224 +242,235 @@ def lst_func(parser: argparse.ArgumentParser) -> None:
-	if sys.argv.__len__() == 1:
-		parser.print_help()
-		return
+    list = subparser.add_parser("list", help="list wordlists")
```
`list` is an [inbuilt data-type](https://docs.python.org/3/tutorial/datastructures.html#more-on-lists). Do not override.

Sable-20 replied:
fixed

Sable-20 committed:
fixed accidental override 

overrode something similar forgot i changed name for reading and went to change it back

pwnfoo commented on wordlistctl.py:
```diff
@@ -1,303 +1,476 @@
+            return str.join("\n", output)
+
+        def print_usage(self, file=None):
+            if (file == None):
```
```python
if not file:
```

Kill brackets

Sable-20 committed:
redundant ()

pwnfoo commented on wordlistctl.py:
```diff
@@ -267,12 +268,12 @@ def add_argument(self, *args, **kwargs):
                 wrapper.initial_indent = prefix
                 wrapper.subsequent_indent = len(prefix) * " "
-                if (self.program["usage"] == "" or str.isspace(self.program["usage"])):
+                if self.program["usage"] == "" or str.isspace(self.program["usage"]):
```
If you are trying to kill trailing/leading whitespaces, you can just `strip()` the instance instead of this check. If not : 

```python
if not self.program["usage"].replace(" ",""):
```

Sable-20 replied:
kind of done i guess

Sable-20 committed:
got rid of the last of the ()

Sable-20 commented:
I forgot this was still open thought it was merged, all seen issues should be fixed
